magicctapipe/scripts/lst1_magic/lst1_magic_cloud_correction.py

```diff
@@ -0,0 +1,715 @@
+            "time_diff": time_diff,
+            "n_pixels": signal_pixels,
+            "n_islands": n_islands,
+            "event_id_lst": event_id_lst,
+            "obs_id_lst": obs_id_lst,
+            "event_id_magic": event_id_magic,
+            "obs_id_magic": obs_id_magic,
+            "combo_type": combo_type,
+            "timestamp": timestamp,
+            "multiplicity": multiplicity,
+        }
+        event_params.update(event_info_dict)
+
+        all_params_list.append(event_params)
+
+    df = pd.DataFrame(all_params_list)
+    return df
+
+
+def main():
+    """Main function."""
+    start_time = time.time()
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument(
+        "--input_file",
+        "-i",
+        dest="input_file",
+        type=str,
+        required=True,
+        help="Path to an input .h5 DL1 data file",
+    )
+
+    parser.add_argument(
+        "--output_dir",
+        "-o",
+        dest="output_dir",
+        type=str,
+        default="./data",
+        help="Path to a directory where to save an output corrected DL1 file",
+    )
+
+    parser.add_argument(
+        "--config_file",
+        "-c",
+        dest="config_file",
+        type=str,
+        default="./resources/config.yaml",
+        help="Path to a configuration file",
+    )
+
+    args = parser.parse_args()
+
+    subarray_info = SubarrayDescription.from_hdf(args.input_file)
+
+    tel_descriptions = subarray_info.tel
+    camgeom = {}
+    for telid, telescope in tel_descriptions.items():
+        camgeom[telid] = telescope.camera.geometry
+
+    optics_table = read_table(
+        args.input_file, "/configuration/instrument/telescope/optics"
+    )
+    focal_eff = {}
+
+    for telid, telescope in tel_descriptions.items():
+        optics_row = optics_table[optics_table["optics_name"] == telescope.name]
+        if len(optics_row) > 0:
+            focal_eff[telid] = optics_row["effective_focal_length"][0] * u.m
+        else:
+            raise ValueError(f"No optics data found for telescope: {telescope.name}")
+
+    with open(args.config_file, "r") as file:
+        config = yaml.safe_load(file)
+
+    tel_ids = config["mc_tel_ids"]
+
+    correction_params = config.get("cloud_correction", {})
+    max_gap_lidar_shots = u.Quantity(
+        correction_params.get("max_gap_lidar_shots")
+    )  # set it to 900 seconds
+    lidar_report_file = correction_params.get(
+        "lidar_report_file"
+    )  # path to the lidar report
+    nlayers = correction_params.get("number_of_layers")
+    cmf = correction_params.get("cleaning_multiplication_factor")
+
+    dl1_params = read_table(args.input_file, "/events/parameters")
+
+    mean_subrun_timestamp = np.mean(dl1_params["timestamp"])
+    mean_subrun_zenith = np.mean(90.0 - np.rad2deg(dl1_params["pointing_alt"]))
+
+    cloud_params = lidar_cloud_interpolation(
+        mean_subrun_timestamp, max_gap_lidar_shots, lidar_report_file
+    )
```

Comment:
this function can return None if there are no proper LIDAR raports to interpolate from that would crash the code below. A better way to handle error management would be to introduce a new error here:
https://github.com/cta-observatory/magic-cta-pipe/blob/73e4a3bf47ace47deb11d1d74a4caf0c37954841/magicctapipe/utils/error_codes.py
and throw it like here: https://github.com/cta-observatory/magic-cta-pipe/blob/73e4a3bf47ace47deb11d1d74a4caf0c37954841/magicctapipe/scripts/lst1_magic/lst1_magic_stereo_reco.py#L234